src/main/java/com/petproj/moneychecker/entities/Category.java

```diff
@@ -0,0 +1,36 @@
+package com.petproj.moneychecker.entities;
+
+import jakarta.persistence.*;
+import lombok.*;
+
+import java.util.Set;
+
+
+@NoArgsConstructor
+@AllArgsConstructor
+@Getter
+@Setter
+@EqualsAndHashCode(of = {"id"})
+@ToString
+@Entity
+@Table(name = "categories", schema = "public")
+public class Category {
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.AUTO)
+    @Column(name = "category_id")
+    private long id;
+    @Column(name = "name")
+    private String name;
+    @Column(name = "description")
+    private String description;
+
+    @OneToMany(mappedBy = "category")
+    private Set<Expense> expenseSet;
+
+    @ManyToOne
+    @JoinColumn(name = "user_id")
+    private User user;

```

Comment:
All categories can be bound to many users, and the opposite works as well - many users can have many categories. So switch to many-to-many